sensor/kubernetes/listener/resources/convert.go

```diff
@@ -362,11 +373,41 @@ func (w *deploymentWrap) populateImageMetadata(localImages set.StringSet, pods .
 				image.Id = digest
 				image.NotPullable = !imageUtils.IsPullable(c.ImageID)
 				image.IsClusterLocal = localImages.Contains(image.GetName().GetFullName())
+				updateImageWithNewerImageName(image, runtimeImageName, false)
 			}
 		}
 	}
 }
 
+// updateImageWithNewerImageName will update the registry, remote, and full name
+// of image from a newer image name, such as the one provided by the container
+// runtime. This is needed in order to support CRI-O's unqualified search registries
+// and short name aliases, for more details refer to:
+// https://github.com/containers/image/blob/main/docs/containers-registries.conf.5.md
+//
+// The image in a workload's spec may differ from what the container runtime pulls,
+// therefore the image name must be updated to accurately reflect the pulled image
+// to give downstream processes (like scanning) a better chance at completing.
```

Comment:
Is it worth mentioning this does not account for/handle mirror registries (along with a link/details for how/where we handle that) or would that just add confusion?

Reply:
I'm torn on that, I'm 'hoping' that in the future this field will represent a mirror as well, in which case we would miss updating the comment (causing confusion) or on the flip side it could cause confusion now... since you asked I'll add a note about mirroring. 😉